proxy/src/bind.rs

```diff
@@ -260,6 +273,23 @@ where
         Reconnect::new(proxy)
     }
 
+    /// Bind the _whole_ stack, including the `WatchService` that rebinds
+    /// the inner stack on TLS client configuration changes.
```

Comment:
This documentation assumes you know how this is implemented internally, which defeats the purpose of the docstring imo.  A helpful doc would include a description of what you mean by "whole stack", for instance.  Similarly it refers to "the WatchService", but doesn't explain at all what it does for us or why we care about it.

I'd prefer to make this much more general and explanatory, or leave it out.

Reply:
Yeah, the comments here were somewhat rushed. Will try to make them more useful.

Reply:
(as a side note, part of why these comments are so implementation-focused is because they're on _private_ functions rather than public ones, so I figured anyone reading them was reading the internal implementation...)